### PR TITLE
Update docs with CI integration support info

### LINKED ISSUE #414: Create Robocop github action
Create separate Robocop github actions so others can easily define workflow with our tool, ie:
```
- name: Robocop lint
  uses: MarketSquare/actions-robocop@master
```

First version should only support configuraton from file. For github annotations we will need ProblemMatcher file like:
https://github.com/ammaraskar/gcc-problem-matcher/blob/master/gcc_matcher.json
https://github.com/korelstar/xmllint-problem-matcher/blob/master/xmllint-matcher.json
https://github.com/codespell-project/actions-codespell/blob/master/codespell-problem-matcher/codespell-matcher.json

It would be best if Robocop could generate ProblemMatcher file when run with special option (because this file will change depending on configured format).

### PR DESCRIPTION
**Pull request by bhirsz**
My spelling/grammar probably will require redo of some parts but I've wanted to include this information in our docs. 
I can try to recreate Jenkins example soon to see if I can provide more detailed intruction.

Closes #414 because thanks for support SARIF file we don't need to create our own Github action - everyone can creater theirs and use output from Robocop to register issues.

### CODE REVIEW

File: docs/ci.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
+.. _ci:
+
+Continous Integration support
```
Comment (mnojek) [inline docs/ci.rst:3]:
```suggestion
Continuous Integration support
```

File: docs/ci.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+=========================================
+
+You can use Robocop in CI/CD from any provider using generated reports or output file.
+
+If there is no direct support for the Robocop, you can raise an issue to add support for it
+(in our `issue tracker <https://github.com/MarketSquare/robotframework-robocop/issues>`_) or implement your own solution.
+
+One of the important configuration for CI integration is return status.
```
Comment (mnojek) [inline docs/ci.rst:11]:
```suggestion
One of the important configurations for CI integration is 'return status'.
```

File: docs/ci.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+See the `docs <https://robocop.readthedocs.io/en/stable/user_guide.html#return-status>`_.
+
+Github Code Scanning
+----------------------
+You can integrate Robocop results with Github Code Scanning (`Github documentation <https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning>`_).
+It is possible using SARIF (Static Analysis Results Interchange Format) output format. Example below shows Github Workflow that runs the Robocop
```
Comment (mnojek) [inline docs/ci.rst:17]:
```suggestion
It is possible to use SARIF (Static Analysis Results Interchange Format) output format. Example below shows Github Workflow that runs the Robocop
```

File: docs/ci.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+.. image:: images/github_code_scanning1.png
+  :alt: Code Scanning in PR
+
+
+Issue details:
+
+.. image:: images/github_code_scanning2.png
+  :alt: Code Scanning issue details
+
+You can configure the Robocop using cli or configuration file (:ref:`configuration file`).
```
Comment (mnojek) [inline docs/ci.rst:60]:
```suggestion
You can configure Robocop using CLI or configuration file (:ref:`configuration file`).
```

File: docs/ci.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+Jenkins
+----------
+There is no direct support for Robocop in the Jenkins. But it is possible to use existing plugins for tools like
```
Comment (mnojek) [inline docs/ci.rst:64]:
```suggestion
There is no direct support for Robocop in Jenkins. But it is possible to use existing plugins for tools like
```

**Review by mnojek — COMMENTED** [review]
Small suggestions, but overall it looks great!

**Review by mnojek — APPROVED** [review]

**Comment by ds-dustenharrison** [comment]
Looks like it is not quite working. Here is what I have in my github action:
       python -m robocop --reports sarif.json .
However, when I look at the directory the sarif file is not there
Run pwd
  pwd
  ls -lha
  shell: /usr/bin/bash -e {0}
  shell: /usr/bin/bash -e {0}
/home/runner/work/qa-testing-robocop/qa-testing-robocop
total 64K
drwxr-xr-x 10 runner docker 4.0K Aug [2](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:2)4 18:22 .
drwxr-xr-x  [3](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:3) runner docker [4](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:4).0K Aug 24 18:22 ..
drwxr-xr-x  8 runner docker 4.0K Aug 24 18:22 .git
drwxr-xr-x  3 runner docker 4.0K Aug 24 18:22 .github
drwxr-xr-x  3 runner docker 4.0K Aug 24 18:22 .idea
drwxr-xr-x  4 runner docker 4.0K Aug 24 18:22 Data
-rw-r--r--  1 runner docker  17[5](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:5) Aug 24 18:22 Dockerfile
drwxr-xr-x  4 runner docker 4.0K Aug 24 18:22 Library
-rw-r--r--  1 runner docker   80 Aug 24 18:22 README.md
drwxr-xr-x  5 runner docker 4.0K Aug 24 18:22 Resources
drwxr-xr-x  7 runner docker 4.0K Aug 24 18:22 Tests
drwxr-xr-x  2 runner docker 4.0K Aug 24 18:22 Tools
-rw-r--r--  1 runner docker 1.5K Aug 24 18:22 execution_order.dat
-rw-r--r--  1 runner docker  519 Aug 24 18:22 main.py
-rw-r--r--  1 runner docker  29[6](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:7) Aug 24 1[8](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:9):22 requirements.txt
-rwxr-xr-x  1 runner docker  [9](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:10)44 Aug 24 [18](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:19):[22](https://github.com/LiveoakTechnologies/qa-testing-robocop/runs/8001583328?check_suite_focus=true#step:5:23) rfdocker

**Pull request merged**